src/pages/private-route/private-route.tsx

```diff
@@ -21,10 +21,13 @@ export const PrivateRoute = observer(
     component,
   }: ProtectedRouteProps) => {
     if (isAuthenticated) {
-      if (isAdminRoute && isAdmin) {
-        return component;
-      } else {
+      if (isAdminRoute) {
+        if (isAdmin) {
+          return component;
+        }
         return <Navigate to={{ pathname: "/homepage" }} />;
+      } else {
+        return component;
       }
     } else {
       return <Navigate to={{ pathname: authenticationPath }} />;
```

Comment:
The if chaining is too nested, try to divide this entire if block into 3 separate if blocks with the according conditions.